depthai_nodes/parsing_neural_network.py

```diff
@@ -39,22 +39,10 @@ def __init__(self, *args, **kwargs) -> None:
         self._nn = self._pipeline.create(dai.node.NeuralNetwork)
         self._parsers: Dict[int, BaseParser] = {}
 
-    @overload
-    def build(
-        self, input: dai.Node.Output, nn_source: dai.NNModelDescription, fps: int
-    ) -> "ParsingNeuralNetwork":
-        ...
-
-    @overload
-    def build(
-        self, input: dai.Node.Output, nn_source: dai.NNArchive, fps: int
-    ) -> "ParsingNeuralNetwork":
-        ...
-
     def build(
         self,
         input: dai.Node.Output,
-        nn_source: Union[dai.NNModelDescription, dai.NNArchive],
+        model: str,
```

Comment:
I was thinking about this, and remembered we said to keep the same syntax as `dai.NeuralNetwork` node, currently there are 2 accepted build functions: `.build( input: Camera, modelDesc: dai.NNModelDescription, fps: float)`and `.build(input: Camera, nnArchive: dai.NNArchive, fps: float)`. For keeping in sync with depthai syntax/function definitions, I would split the this build method in the same way.

Reply:
cc @klemen1999 on this. We can add overrides and allow the old `nn_source` types. But the option of using slug should stay as it comes quite handy (e.g. when re-writing experiments).

Reply:
Can we keep all three options? Basically add that nn_source can also be str and handle this properly inside the build()?

Reply:
Adding in [5d2208f](https://github.com/luxonis/depthai-nodes/pull/134/commits/5d2208faf8a891d59bea2a82b39db51352f2b69f).